File: 2. Loop Problems/Loop Problems in c/p17.c
// .Write a C program to check whether a number is Prime number or not


#include<stdio.h>
#include<math.h>

int main () {
    int n;
    printf("Enter Your Number: ");
    scanf("%d",&n);

    if (n <= 1) {
        printf("%d is not a Prime.",n);
        return 0;
    }

    if (n == 2 || n == 3  ) {
        printf("%d is a Prime.",n);
        return 1;
    }

    if (n % 2 == 0 || n % 3 == 0) {
        printf("%d is not a Prime.",n);
        return 0;
    }

    for (int i = 5;i <= sqrt(n);i+=6) {
         if (n % i == 0 || n% (i + 2) == 0) {
                printf("%d is not a Prime.",n); 
                break; 
    }

    
}
printf("%d is a Prime.",n);

}

File: 2. Loop Problems/Loop problems in python/p17.py
# // Write a python program to check whether a number is Prime number or not

import math


n = int(input("Enter Your Number: "))


if n <= 1:
    print(f"{n} is not a Prime.")
elif n == 2 or n == 3:
    print(f"{n} is a Prime.")
elif n % 2 == 0 or n % 3 == 0:
    print(f"{n} is not a Prime.")
else:
    
    is_prime = True  # Assume the number is prime
    for i in range(5, int(math.sqrt(n)) + 1, 6):
        if n % i == 0 or n % (i + 2) == 0:
            print(f"{n} is not a Prime.")
            is_prime = False
            break

  
   
    if is_prime:
        print(f"{n} is a Prime.")
